presto-spi/src/main/java/com/facebook/presto/spi/relation/RowExpression.java

```diff
@@ -66,4 +66,6 @@ public Optional<SourceLocation> getSourceLocation()
     public abstract String toString();
 
     public abstract <R, C> R accept(RowExpressionVisitor<R, C> visitor, C context);
+
+    public abstract RowExpression withoutSourceLocation();
```

Comment:
maybe just call it `canonicalize`. Whose know if someday we gonna add more non-critical information to this so we need to strip it out again.

Also, add a javadoc to it.